### DISCUSSION TITLE
Foundryup command not working

### DISCUSSION THREAD

**Discussion by Killpit** [Cyfrin/security-and-auditing-full-course-s23#43] [Q&A]
Hi there, 

Even though I have foundry and its dependencies, whenever I said foundryup, it says the command is not found. Tried installing through different methods and still failed to get it working. So my plan is to remove foundry entirely and then re-installing it, I don't really know how I can make foundry work.

For foundry, I tried curl and cargo commands and had the same problem, tried installing from the same source but still foundryup command didn't work

**Comment by akhilmanga** [comment]
Try to install Foundry in the correct directory.

**Reply by Killpit** [reply]
Actually, I've followed everything the documentation told me, and what is the correct repo?

And here's the issue for you to examine it closer:

https://github.com/foundry-rs/foundry/issues/6668#issuecomment-1870318897

**Reply by PatrickAlphaC** [reply]
Can you try working with ChatGPT to debug?

**Reply by Killpit** [reply]
Used Bard, but looks like user config problem, will try to remove duplicates and see how that goes.

**Reply by Killpit** [reply]
Will try foundry again at your course, did one try and it worked now, have to check paths

**Comment by EngrPips** [comment]
@Killpit  are you all good with foundry now ?

**Comment by ibanigabriel** [comment]
Hi 

Please I don't have a foundry and am trying to create one on gitbash how do I do it please

**Reply by EngrPips** [reply]
Hello @ibanigabriel, This https://book.getfoundry.sh/getting-started/installation link will help you install the foundry.

**Comment by cas443** [comment]
Ah, this is an easy fix. The issue is that when you added foundry to your PATH, through `export PATH=$PATH:/Users/atatekeli/.foundry/bin`, that only adds it to your path *locally* meaning that if you open a new terminal window or try to run the command through VSCode, it's not going to recognize it.

The fix is to add it to your path *globally* by adding an entry in your `~/.zshrc` (or the equiv. to whichever shell you're using). At the end of the file, just append the following: `export PATH="$PATH:/Users/atatekeli/.foundry/bin"`. Then run `exec zsh` (or the equiv. to whichever shell you're using) to apply the changes to current terminal window.

You should be able to run foundry from any terminal window now.

**Reply by EngrPips** [reply]
Hey @cas443, when you said it adds it to the path locally, you meant it adds it to the current working directory, right? I don't know how to add an entry in my `~/.zshrc` can you please inform me on that.

**Reply by cas443** [reply]
Not so much the current working directory but rather it makes the path accessible in the current terminal session (tab) you have open.

Writing to `~/.zshrc` is really simple, you can do it with following one liner:
`echo 'export PATH=$PATH:/new/directory/path' >> ~/.zshrc`

Keep in mind that the file you're writing to may not necessarily be `~/.zshrc`, it could be `~/.bashrc` or `~/.bash_profile` etc. depending on the particular shell you're using.

EDIT:

Remember to run `source ~/.bashrc` (or the equiv. to whichever shell you're using) for the changes to take effect after updating the file.

**Reply by EngrPips** [reply]
Thanks for this response. It helps to clarify some stuff.
